Gems/EMotionFX/Code/Source/Integration/Components/SimpleLODComponent.cpp

```diff
@@ -201,15 +208,24 @@ namespace EMotionFX
                 AZ::RPI::ViewportContextPtr defaultViewportContext =
                     viewportContextManager->GetViewportContextByName(viewportContextManager->GetDefaultViewportContextName());
                 const float distance = worldPos.GetDistance(defaultViewportContext->GetCameraTransform().GetTranslation());
-                const size_t lodByDistance = GetLodByDistance(configuration.m_lodDistances, distance);
-                actorInstance->SetLODLevel(lodByDistance);
+                const size_t requestedLod = GetLodByDistance(configuration.m_lodDistances, distance);
+                actorInstance->SetLODLevel(requestedLod);
 
                 if (configuration.m_enableLodSampling)
                 {
-                    const float animGraphSampleRate = configuration.m_lodSampleRates[lodByDistance];
+                    const float animGraphSampleRate = configuration.m_lodSampleRates[requestedLod];
                     const float updateRateInSeconds = animGraphSampleRate > 0.0f ? 1.0f / animGraphSampleRate : 0.0f;
                     actorInstance->SetMotionSamplingRate(updateRateInSeconds);
                 }
+
+                // When setting the actor instance LOD level, a change is just requested and with the next update it will get applied.
+                // This means that the current LOD level might differ from the requested one. We need to sync the Atom LOD level with the
+                // current LOD level of the actor instance to avoid skinning artifacts. The requested LOD level will be present and applied
+                // the following frame.
+                const size_t currentLod = actorInstance->GetLODLevel();
+                AZ::Render::MeshComponentRequestBus::Event(entityId,
+                    &AZ::Render::MeshComponentRequestBus::Events::SetLodOverride,
+                    static_cast<AZ::RPI::Cullable::LodOverride>(currentLod));
```

Comment:
I moved it inside the game loop directly. The check would result in an ebus call, and if the type differs we'd need a second call, so I directly added the set call in there.

The long-term solution for this is retrieving the lod level from the visibility system, but that is not there yet. The issue currently is that when having multiple render targets, there might be several geometry LOD levels set on different targets while emfx only updates the skeleton once. So we need to update it with the lowest LOD level from the render targets. The problem is that emfx updates earlier in the pipe and there is no way to query that at the moment. So we need to take ownership and control the lod level as a short- to mid-term solution until the new LOD system is in place.

Reply:
I'm not sure AZ::Render::MeshComponentRequestBus::Events::SetLodType need to get called every frame.
From my understanding from the conversation - simpleLODComponent, if present, will take ownership of the mesh LOD (which override the any lod culling method from atom mesh component internal)
It could be done once immediately after the actor is ready to use.

Reply:
Roman is correct, it does not need to be called every frame. You could just call the bus events whenever the lod changes instead of every frame. It is a cheap operation, so not particularly harmful to do it every frame. It just sets the values (doesn't have to rebuild the culling data or anything), so the ebus event is probably the most expensive part. So it should not be prohibitive to do it every frame, and that is simpler than keeping track of both when the lod changes by distance and when the actorInstance 'request' is actually applied.